src/modules/cpu/kernel/noise_shot.hpp

```diff
@@ -0,0 +1,1666 @@
+#include "rppdefs.h"
+#include "rpp_cpu_simd.hpp"
+#include "rpp_cpu_common.hpp"
+
+RppStatus shot_noise_u8_u8_host_tensor(Rpp8u *srcPtr,
```

Comment:
This file is too large with very similar code in many functions. We need better code so it won't replicate code in all the functions

Reply:
A couple hundred lines were reduced in those files through this change - https://github.com/GPUOpen-ProfessionalCompute-Libraries/rpp/pull/120#discussion_r947390200